examples/cat_facts/shared_types/build.rs

```diff
@@ -15,9 +15,13 @@ fn main() -> anyhow::Result<()> {
 
     gen.register_app::<CatFacts>()?;
 
+    // types from `crux_http` that aren't automatically discovered
     gen.register_type::<HttpError>()?;
+
+    // types from `crux_kv` that aren't automatically discovered
     gen.register_type::<KeyValueResponse>()?;
     gen.register_type::<KeyValueError>()?;
+    gen.register_type::<Value>()?;
```

Comment:
It may be beating a dead horse with a stick, but we could extend the Capability trait to support automatic registration of this kind of types - that way the auto discovery can explicitly _ask_ the capabilities to register types 🤔

Reply:
I like this idea. I guess it depends on how long it takes me to implement new typegen (could be a while), because then we'd need to take it out again. Let's chat about it. Either way, another PR.

Reply:
I started an attempt at this very idea a while back, as an attempt to fix https://github.com/redbadger/crux/issues/222.

Forget why I didn't finish it, possibly just got bored 😅